ml-agents/mlagents/trainers/trainer.py

```diff
@@ -19,14 +19,15 @@ class UnityTrainerException(UnityException):
 class Trainer(object):
     """This class is the abstract class for the mlagents.trainers"""
 
-    def __init__(self, brain_name, trainer_parameters, training, run_id):
+    def __init__(self, brain, trainer_parameters, training, run_id):
         """
         Responsible for collecting experiences and training a neural network model.
         :param trainer_parameters: The parameters for the trainer (dictionary).
         :param training: Whether the trainer is set for training.
         :param run_id: The identifier of the current run
         """
-        self.brain_name = brain_name
+        self.TRAINER_NAME = 'Base'
```

Comment:
From what I understand this `TRAINER_NAME` is used mainly during logging an exception? If this is correct, can we use the `.__class__` method of the object to determine what class generated this and put that in the exception? I think this will ensure that any future refactors of the class won't lead to the two class name and `TRAINER_NAME` being robust to desynchronization.